### ISSUE TITLE
lifecycleOwner  property with databinding auto generated models

### ISSUE THREAD

**Issue by SBD580** [airbnb/epoxy#412]
Hi,

Thanks for this great library!

Having an issue after upgrading to Android Studio 3.1 which introduce the Android Architecture Components. If my layout includes another layout the `setLifecycleOwner` method is being overridden in the generated binding class and then being treated as an attribute. This is causing a build time exception while the generated (*Model_) class is trying to use `BR.lifecycleOwner` which doesn't really exists.

**Comment by elihart** [comment]
Thanks for reporting. Can you provide an example layout with the issue, as well as the generated model code that is wrong?

**Comment by SBD580** [comment]
Sure, here you go:

activity_main.xml
```
<?xml version="1.0" encoding="utf-8"?>
<layout xmlns:android="http://schemas.android.com/apk/res/android">
    <data />
    <android.support.constraint.ConstraintLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent">
        <include layout="@layout/included_layout" />
    </android.support.constraint.ConstraintLayout>
</layout>
```

included_layout.xml
```
<?xml version="1.0" encoding="utf-8"?>
<layout xmlns:android="http://schemas.android.com/apk/res/android">

    <data />

    <android.support.constraint.ConstraintLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent" />
</layout>
```

build.gradle
```
apply plugin: 'com.android.application'

android {
    compileSdkVersion 27
    defaultConfig {
        applicationId "com.sbd580.eproxyissue412"
        minSdkVersion 21
        targetSdkVersion 27
        versionCode 1
        versionName "1.0"
        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
    }
    dataBinding {
        enabled true
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
}

ext.set("supportVersion", '27.1.0')
//
configurations.all {
    resolutionStrategy {
        eachDependency { details ->
            // Force all of the primary support libraries to use the same version.
            if (details.requested.group == 'com.android.support'
                    && details.requested.name != 'multidex'
                    && details.requested.name != 'multidex-instrumentation') {
                details.useVersion supportVersion
            }
        }
    }
}

dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    implementation "com.android.support:appcompat-v7:$supportVersion"
    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
    testImplementation 'junit:junit:4.12'
    androidTestImplementation 'com.android.support.test:runner:1.0.1'
    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'

    implementation 'com.airbnb.android:epoxy:2.10.0'
    implementation 'com.airbnb.android:epoxy-databinding:2.10.0'
    annotationProcessor 'com.airbnb.android:epoxy-processor:2.10.0'
}
```

package-info.java
```
@EpoxyDataBindingLayouts({R.layout.activity_main})
@PackageModelViewConfig(rClass = R.class)
package com.sbd580.eproxyissue412;

import com.airbnb.epoxy.EpoxyDataBindingLayouts;
import com.airbnb.epoxy.PackageEpoxyConfig;
import com.airbnb.epoxy.PackageModelViewConfig;
import com.sbd580.eproxyissue412.R;
```

(GENERATED) ActivityMainBindingModel_.java
```
.....

  @Override
  protected void setDataBindingVariables(ViewDataBinding binding) {
    if (!binding.setVariable(BR.lifecycleOwner, lifecycleOwner)) {
      throw new IllegalStateException("The attribute lifecycleOwner was defined in your data binding model (com.airbnb.epoxy.DataBindingEpoxyModel) but a data variable of that name was not found in the layout.");
    }
  }

  @Override
  protected void setDataBindingVariables(ViewDataBinding binding, EpoxyModel previousModel) {
    if (!(previousModel instanceof ActivityMainBindingModel_)) {
      setDataBindingVariables(binding);
      return;
    }
    ActivityMainBindingModel_ that = (ActivityMainBindingModel_) previousModel;
    if (((lifecycleOwner == null) != (that.lifecycleOwner == null))) {
      binding.setVariable(BR.lifecycleOwner, lifecycleOwner);
    }
  }
...
```

(sample repo: https://github.com/SBD580/epoxy-issue-412)

**Comment by elihart** [comment]
Thanks!

Epoxy just looks at all of the setters in the generated binding class, it should be easy to blacklist `setLifecycleOwner`.

@ngsilverman is this something you would be interested in fixing? should be pretty easy

**Comment by elihart** [comment]
Thanks everybody that put up a fix. I have merged one and am pushing it in release 2.11.0 right now. Should be live in 30 minutes or so.
